includes/admin/class-wc-rest-payments-webhook-controller.php

```diff
@@ -301,15 +301,16 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 		// Get an updated set of order properties to avoid race conditions when the server sends the paid webhook before we've finished processing the original payment request.
 		$order->get_data_store()->read( $order );
 
-		// prevent parallel order processing attempts.
-		if ( WC_Payments_Utils::lock_order_payment( $order, $intent_id ) ) {
+		if ( $order->has_status( [ 'processing', 'completed' ] ) ) {
```

Comment:
This check is no longer needed with locks introduced in this PR but in some cases, this would allow us to bail early without having to place locks, so decided to keep it.